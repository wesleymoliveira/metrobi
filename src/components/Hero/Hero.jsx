import { Typography } from '../Typography/Typography';
import * as S from './Hero.styled';

export function Hero() {
	return (
		<S.Wrapper data-testid='hero'>
			<Typography size='xxlarge'>
				Hero
			</Typography>
		</S.Wrapper>
	);
}
