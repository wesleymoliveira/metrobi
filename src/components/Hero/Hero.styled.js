import styled, { css } from 'styled-components'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    background-color: ${theme.colors.heroBackground};
    flex: 2;
    display: flex;
    justify-content: center;
    align-items: start;
  `}
`;
