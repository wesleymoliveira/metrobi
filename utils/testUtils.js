import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
