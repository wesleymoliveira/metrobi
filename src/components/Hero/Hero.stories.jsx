import React from 'react';
import { Hero } from './Hero';

const HeroArgTypes = {
  label: {
    name: 'label',
    type: { name: 'string', required: false },
    description: 'overwritten description',
    table: {
      type: { 
        summary: 'something short', 
        detail: 'something really really long' 
      },

    },
    control: {
      type: 'string'
    }
  }
};

export default {
	title: 'components/Hero',
	component: Hero,
	argTypes: HeroArgTypes
};

export const Default = (args) => <Hero {...args} />;
