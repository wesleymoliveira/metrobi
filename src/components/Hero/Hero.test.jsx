import { screen } from '@testing-library/react';

import { Hero } from './Hero';
import { render, snapshotRenderer } from '../../../utils/testUtils';

describe('<Hero />', () => {
		it('should render the component', () => {
		render(<Hero />);
		
		expect(screen.getByTestId(/hero/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<Hero />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should match the original background-color', () => {
		const tree = snapshotRenderer(<Hero />).toJSON();
		//make sure the background-color is the original one
		expect(tree).toHaveStyleRule('background-color', '#d7c9e3')
	});

});
