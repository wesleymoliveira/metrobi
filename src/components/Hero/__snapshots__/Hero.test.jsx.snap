// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should match snapshot 1`] = `
.c1 {
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c0 {
  background-color: #d7c9e3;
  height: 40%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

<section
  className="c0"
  data-testid="hero"
>
  <h1
    className="c1"
    color="white"
    data-testid="typography"
    size="xxlarge"
  >
    Hero
  </h1>
</section>
`;
